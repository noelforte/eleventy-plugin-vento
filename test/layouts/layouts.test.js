import { runBuild } from '../get-instance.js';
import { expect, test } from 'vitest';

const results = await runBuild(import.meta.dirname);

test('render page with example layout 1', () => {
	const page = results.find(({ url }) => url === '/page1/');
	expect(page.content).toMatchSnapshot();
});

test('render page with example layout 2', () => {
	const page = results.find(({ url }) => url === '/page2/');
	expect(page.content).toMatchSnapshot();
});
