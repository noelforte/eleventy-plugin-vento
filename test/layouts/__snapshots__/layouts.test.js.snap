// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render page with example layout 1 1`] = `
"<!doctype html>
<html lang="en">
	<head>
		<meta charset="UTF-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
		<title>Page with layout</title>
	</head>
	<body>
		<header>
			<h1>Page title: Page with layout</h1>
		</header>
		<main>
			<section>This is an alternative layout.</section>
			<section>
<p>This is a page that uses the first layout.</p>

<p>This is another paragraph.</p>
<p>This is another paragraph.</p>
<p>This is another paragraph.</p>

</section>
		</main>
		<footer>
			<p>I am a footer, outside the layout</p>
		</footer>
	</body>
</html>
"
`;

exports[`render page with example layout 2 1`] = `
"<!doctype html>
<html lang="en">
	<head>
		<meta charset="UTF-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
		<title>Page with layout</title>
	</head>
	<body>
		<header>
			<h1>Page title: Page with layout</h1>
		</header>
		<main>
<p>This is a page that uses the second layout.</p>

<p>This is another paragraph.</p>
<p>This is another paragraph.</p>
<p>This is another paragraph.</p>

</main>
		<footer>
			<p>I am a footer, outside the layout</p>
		</footer>
	</body>
</html>
"
`;
